ecosystem/purescript-0.14-migration-guide.md

```diff
@@ -0,0 +1,622 @@
+
+```purs
+-- Code written pre-PureScript 0.14, which will compile with PureScript 0.14
+-- but will not compile with PureScript 0.15.
+getFoo = Record.get (SProxy :: SProxy "foo")
+
+-- Code written for PureScript 0.14 which will continue to work with
+-- PureScript 0.15.
+getFoo = Record.get (Proxy :: Proxy "foo")
+```
+
+
+#### The `generics-rep` library has been merged into `prelude`
+
+`Generic` has been moved to the bottom of the core ecosystem because it's widely-used class with compiler support, but it didn't previously cover some common types like `Tuple` or `Either`. This change enables all of the data types across the core libraries which have a sensible `Generic` instance to have one.
+
+**To fix:**
+
+You should remove `generics-rep` as a dependency of your library or application, as most of the relevant code and instances live in the `prelude` package. Code related to `Enum` has been moved to the `enums` package, but you should already depend on this package if you were using it.
+
+You should rename any modules that used `Generic` instances for common type classes, as those modules have been renamed.
+
+```txt
+Data.Generic.Rep.Bounded        -> Data.Bounded.Generic
+Data.Generic.Rep.Enum           -> Data.Enum.Generic (in the `enums` package)
+Data.Generic.Rep.Eq             -> Data.Eq.Generic
+Data.Generic.Rep.HeytingAlgebra -> Data.HeytingAlgebra.Generic
+Data.Generic.Rep.Monoid         -> Data.Monoid.Generic
+Data.Generic.Rep.Ord            -> Data.Ord.Generic
+Data.Generic.Rep.Ring           -> Data.Ring.Generic
+Data.Generic.Rep.Semigroup      -> Data.Semigroup.Generic
+Data.Generic.Rep.Semiring       -> Data.Semiring.Generic
+Data.Generic.Rep.Show           -> Data.Show.Generic
+```
+
+The `argonaut-generic` library has also renamed its modules to match this naming convention:
+
+```txt
+Data.Argonaut.Decode.Generic.Rep -> Data.Argonaut.Decode.Generic
+Data.Argonaut.Encode.Generic.Rep -> Data.Argonaut.Encode.Generic Data.Argonaut.Types.Generic.Rep  -> Data.Argonaut.Types.Generic
+```
+
+#### The `globals` library has been deprecated
+
+The `globals` library provided various functions and values that are in the global scope in JavaScript. The library has been deprecated in particular because PureScript supports multiple backends besides JavaScript, and the relevant functions have been moved or reimplemented in other libraries.
+
+Specifically:
+
+* Number values and operations like `isFinite`, parsing, and formatting have been moved to the `numbers` package, which itself has moved into the core organization.
+* Integer parsing has been removed, as it is already included in the `integers` package.
+* Encoding and decoding URIs and URI components using JavaScript's builtin `encodeURIComponent` and `decodeURIComponent` have been moved to the new `js-uri` package.
+* Unsafe global functions have been removed altogether.
```

Comment:
Can you remind me which ones these were? Was it just the unsafe variants of the URI encoding ones?

Reply:
Ah, it was basically all of them, wasn't it? https://github.com/purescript-deprecated/purescript-globals/commit/2d56edd41da66639f667ad97e4293db75c835883#diff-c475094f3ca830f19ce57d7b2928870987347b3f28bdb5af290c543ebd2d7afc

Perhaps this is slightly clearer then?
```suggestion
* Unsafe global functions (i.e. those which lived in `Global.Unsafe`) have been removed altogether.
```